### PR TITLE
Add a types declaration for typescript

### PR DESCRIPTION
**Pull request by JoshMiers-FS**
This PR adds better type info for typescript projects. I verified all of the API methods in our sample test app (not linking to private repo here) on both Android and iOS.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  Info: 2, // Default
+  Warn: 3,
+  Error: 4,
+  Assert: 5, // Clamps to Error on Android
+};
+
+interface FullStoryInterface {
+  LogLevel: LogLevel;
+  anonymize(): void;
+  identify(string, Object): void;
```
Comment (gunnnnii) [inline src/index.d.ts:23]: This should probably be using `Record<string, any>`. `Object` is effectively the same as `any`.
F.x. this type checks: `FS.identify('test', 123)`.

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  Log: 0, // Clamps to Debug on iOS
+  Debug: 1,
+  Info: 2, // Default
+  Warn: 3,
+  Error: 4,
+  Assert: 5, // Clamps to Error on Android
+};
+
+interface FullStoryInterface {
+  LogLevel: LogLevel;
```
Comment (gunnnnii) [inline src/index.d.ts:21]: This is a type error. It should be `LogLevel: typeof LogLevel`. This causes the type of `FS.LogLevel[level]` to resolve to any when used.

 [Playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wDIQOYeAN2ABs4BeOAbwCg44tsAuOABgBo4B6DuAYSIEMAtmFRwYEOABFgAIwCu2OCjgBLAPIBlGlNkKmARlbaAksgBmEJgCZ2XHWf5yiMbQHV+UZEwDMR2gFEoKGgmABY-OABBVFRgWCYAVltuPiERMQlA4KglZCjkABNglQKqAF8AbioqFWQYOIcEYDgAMSciDXEoAE9TeqhG5upaejxCIiZRgmIqkZwx4ismGG6wYAgzOnnpoiqy6oLgBAEoZqQ0eBaNJjaiDq7euob+Jqqqc-Q4IhwyVo0AOim40BPzs-GQ3XeKE+32wVl+VxBuB2ViRnG4yDkghkcSAA)

Comment (JoshMiers-FS) [reply src/index.d.ts:21]: Thanks! I had this as a TODO to fix up and will try out your suggestion.

File: src/index.d.ts (hunk 1 above)
Comment (JoshMiers-FS) [reply src/index.d.ts:23]: I will try this out. I was following https://reactnative.dev/docs/native-modules-android#argument-types and then needed to go back and fill out the TS-specific types. I appreciate the comment and will see if it works in our internal tests.

File: src/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  identify(string, Object): void;
+  setUserVars(Object): void;
+  onReady(): Promise<OnReadyResponse>;
+  getCurrentSession(): Promise<string>;
+  getCurrentSessionURL(): Promise<string>;
+  consent(boolean): void;
+  event(string, Object);
+  shutdown(): void;
+  restart(): void;
+  log(number, string): void;
```
Comment (nfriedly) [inline src/index.d.ts:32]: Should this be `LogLevel` instead of `number`?

File: src/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+declare const {FullStory} = NativeModules;
+
+export type OnReadyResponse = {
+  replayStartUrl: string;
+  replayNowUrl: string;
+  sessionId: string;
+};
+
+export const LogLevel = {
```
Comment (nfriedly) [inline src/index.d.ts:11]: relatedly, should this be an enum?

File: src/index.d.ts (hunk 3 above)
Comment (nfriedly) [reply src/index.d.ts:32]: Looking at https://reactnative.dev/docs/native-modules-android#argument-types, maybe not...

**Review by nfriedly — APPROVED** [review]
I'd be interested to know if an enum worked here, but either way, I think this looks fine.

(I *think* that if an enum would work here, changing it would be a backwards-compatible change.)

**Comment by JoshMiers-FS** [comment]
> I'd be interested to know if an enum worked here, but either way, I think this looks fine.
> 
> (I _think_ that if an enum would work here, changing it would be a backwards-compatible change.)

It doesn't prevent passing in other types, so I am going to leave it as-is for now and look into further ways to change it.

**Pull request merged**